packages/valory/skills/transaction_settlement_abci/behaviours.py

```diff
@@ -370,27 +370,31 @@ def async_act(self) -> Generator:
 
     def has_transaction_been_sent(self) -> Generator[None, None, Optional[bool]]:
         """Transaction verification."""
+
+        to_be_validated_tx_hash = self.synchronized_data.to_be_validated_tx_hash
+        if not to_be_validated_tx_hash:
+            return False
```

Comment:
So we enter validation without a to_be_validated_tx_hash? Doesnt that mean that we have a problem in our logic then if we reach this situation?

Reply:
Exactly, this is why we were raising before, but we need a better solution because we should only raise in the abstract abci level.

Reply:
now that it is strict, there is no longer the need to transition with an event `NOTHING_TO_VALIDATE` as an error will be raised. Hence I've removed this transition again in a7a767b